lib/pbench/server/filetree.py

```diff
@@ -243,6 +242,59 @@ def create(cls, tarball: Path, controller: "Controller") -> "Tarball":
 
         return cls(destination, controller)
 
+    def extract(self, path: Path) -> str:
+        """
+        Extract a file from the tarball and return it as a string
+
+        TODO: Should we allow fetching a full directory path, e.g., "/" to
+        extract everything? And, if so, in what form would we return it.
+        Similarly, what about a binary file?
```

Comment:
My instinct is that, if someone wants multiple files extracted, then they are going to want a list (or a `dict`!) returned, and we are going to want this function to produce each element in the list...and, therefore, if we want to support extracting multiple files then we should do it with a separate function (which takes a different argument type (a _directory_ `Path`...), which returns a different argument type, and which calls this function as it currently is).  (And, the additional function would likely have additional arguments, e.g., indicating what should be done with subdirectories....)

As for returning a binary file, I think that that should be done as a `bytes` object, not a `str`, but, otherwise, it's not much different from this function.  (And, `bytes` objects and `str`'s are "close enough" (and their respective relationship is well enough understood) that we could _get by_ having this same function return whichever of them is appropriate...which would make implementing the multiple-extract function much simpler to implement.)

Reply:
Yes, we need to architect the "tarball content" metaphor and API. That's part of setting up the "Apache-replacement" `GET` API flowing into our cache management. I'm not going to attack it here, and we've already captured the basics under the object store transition.